src/crypto/tls/handshake_server_tls13.go

```diff
@@ -377,8 +377,31 @@ func (hs *serverHandshakeStateTLS13) pickCertificate() error {
 		c.sendAlert(alertHandshakeFailure)
 		return err
 	}
+
 	hs.cert = certificate
 
+	if hs.clientHello.delegatedCredentialSupported {
+		if c.config.GetDelegatedCredential != nil {
+			dCred, priv, err := c.config.GetDelegatedCredential(clientHelloInfo(c, hs.clientHello))
+			if err != nil {
+				c.sendAlert(alertInternalError)
```

Comment:
Hmm, should we fail, or just fallback to the certificate?

Reply:
I think it should fail as the previous conditions `hs.clientHello.delegatedCredentialSupported && c.config.GetDelegatedCredential` expect a delegatedCredential and not the certificate only.. what do you think?

Reply:
I'm not sure. This seems too fatal. I could imagine a case where `GetDelegatedCredential` genuinely returns nothing, even though the callback itself is not `nil`, as a way of saying "I can't give you a DC right now, please fall back."

Reply:
Mmm.. well, per the draft 'Clients that receive invalid delegated credentials MUST terminate the connection with an "illegal_parameter" alert.'.. does that also include saying that you support DC but receiving nothing?

Reply:
That seems to be saying something different. In particular, it says, if an endpoint receives and tries to validate a DC, but fails, it should produce an alert. However, this code, as I understand it, produces an error if the *sender* can't produce a valid DC. If that's indeed what's happening, I think it's too extreme, and I would prefer we fall back to the normal certificate path. I could see scenarios where a sender can't produce a DC, e.g., if the DC minting code fell over, or something, but the EE certificate is just fine.

Reply:
I changed it now ;)